packages/pancake-liquidity-widgets/src/hooks/useMarketPrice.ts

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useMemo } from "react";
+import {
+  chainIdToChain,
+  NATIVE_TOKEN_ADDRESS,
+  NetworkInfo,
+  PATHS,
+} from "@/constants";
+import { useWeb3Provider } from "@/hooks/useProvider";
+import { PancakeTokenAdvanced } from "@/types/zapInTypes";
+
+export default function useMarketPrice({
+  tokensIn,
+  setTokensIn,
+}: {
+  tokensIn: PancakeTokenAdvanced[];
+  setTokensIn: (value: PancakeTokenAdvanced[]) => void;
```

Comment:
this confuse me, why a hook named `useMarketPrice` can `setTokensIn`?
If I write this hook:
- input: tokens (or tokenAddresses) 
- output: 
    + loading 
    + prices 
    + refetch
    
  Then if you want to enrich token data, the logic should be in the component where consuming this hook

Reply:
or if you want to enrich price info here, I suggest:
- create a separate hook to get prices only: useMarketPrice hook
- rename this to: useEnrichPriceInfo